src/components/TodoItem.tsx

```diff
@@ -0,0 +1,197 @@
+  updateTodo: (par: Todo) => Promise<void>;
+  searchCompletedTodos: () => void;
+  makeTodosComplete: MakeTodosCompleted;
+  clearCompleted: boolean;
+  setClearCompleted: (par: boolean) => void;
+  setMakeTodosComplete: (par: MakeTodosCompleted) => void;
+};
+
+export const TodoItem: FC<Props> = ({
+  todo,
+  deleteTodo,
+  updateTodo,
+  searchCompletedTodos,
+  makeTodosComplete,
+  clearCompleted,
+  setClearCompleted,
+  setMakeTodosComplete,
+}) => {
+  const [checked, setChecked] = useState(todo.completed);
+  const [isClicked, setIsClicked] = useState(false);
+  const [changedTitle, setChangedTitle] = useState(todo.title);
+  const [isDoubleClicked, setIsDoubleClicked] = useState(false);
+
+  const handleComplete = () => {
+    setIsClicked(true);
+    setChecked(!checked);
+    updateTodo({ ...todo, completed: !todo.completed })
+      .then(() => setIsClicked(false));
+    searchCompletedTodos();
+  };
+
+  const handleTitle = (e: FormEvent) => {
+    e.preventDefault();
+    const trimmedChandedTitle = changedTitle.trim();
+
+    if (trimmedChandedTitle === changedTitle.trim()) {
+      setChangedTitle(trimmedChandedTitle);
+    }
+
+    if (!trimmedChandedTitle.length) {
+      setIsClicked(true);
+      deleteTodo(todo.id)
+        .then(() => {
+          setIsDoubleClicked(false);
+        })
+        .catch((error) => {
+          setIsDoubleClicked(true);
+          throw error;
+        })
+        .finally(() => {
+          setIsClicked(false);
+        });
+    }
+
+    if (trimmedChandedTitle.length && trimmedChandedTitle !== todo.title) {
+      setIsClicked(true);
+      updateTodo({ ...todo, title: trimmedChandedTitle })
+        .then(() => {
+          setIsDoubleClicked(false);
+        })
+        .catch(() => {
+          setIsDoubleClicked(false);
+        })
+        .finally(() => {
+          setIsClicked(false);
+        });
+    } else if (trimmedChandedTitle === todo.title) {
+      setIsDoubleClicked(false);
+      setIsClicked(false);
+    } else {
+      setIsClicked(true);
+      setIsDoubleClicked(false);
+    }
+  };
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current && isDoubleClicked) {
+      inputRef.current.focus();
+    }
+  }, [isDoubleClicked]);
+
+  // eslint-disable-next-line
+  useEffect(() => {
+    if (makeTodosComplete === MakeTodosCompleted.do && !checked) {
+      setIsClicked(true);
+      setChecked(true);
+      updateTodo({ ...todo, completed: true })
+        .then(() => setIsClicked(false))
+        .finally(() => setMakeTodosComplete(MakeTodosCompleted.begin));
+    }
+
+    if (makeTodosComplete === MakeTodosCompleted.not && checked) {
+      setIsClicked(true);
+      setChecked(false);
+      updateTodo({ ...todo, completed: false })
+        .then(() => setIsClicked(false))
+        .finally(() => setMakeTodosComplete(MakeTodosCompleted.begin));
+    }
+  }, [makeTodosComplete]);
```

Comment:
update deps

Reply:
i don't need to update depts because if i do it, the code won't work. (we don't need to track another depts, this effect should work only if makeTodosComplete will change